src/components/YouTube.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faYoutube } from '@fortawesome/free-brands-svg-icons';
+
+const YouTubePlayer = ({ youTubeId }) => (
+  <div className="embed-responsive embed-responsive-16by9">
+    <iframe
+      title={youTubeId}
+      className='embed-responsive-item'
+      src={`https://www.youtube.com/embed/${youTubeId}`}
+      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
+      allowFullScreen>
+    </iframe>
+  </div>
+);
+
+class YouTubeToggle extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      visible: false
+    };
+
+    this.handleToggle = this.handleToggle.bind(this);
+  }
+
+  handleToggle() {
+    this.setState((state) => ({
```

Comment:
We should figure out a consistent way to set state. With this we would be doing it two way:

1. `this.setState((state) => { visible: !state.visible });`
2. `this.setState({ visible: !this.state.visible });`

Reply:
Actually, they mean different things. The 2nd one is disallowed because it's modifying the state value of itself. Apparently, that's not good? It's in the doc somewhere. You have to use the 1st one so that the change is deferred until React is ready for the state change to occur (it evaluates the function you pass).

Reply:
Interesting, okay! I'll switch to the first one when I finish up the toolbar.